packages/core/src/Cardano/util/computeImplicitCoin.ts

```diff
@@ -89,67 +103,27 @@ const computeConwayDeposits = (
 };
 
 /** Inspects a transaction for its deposits and returned deposits. */
-const getTxOwnDeposits = (
+const getTxDeposits = (
   { stakeKeyDeposit, poolDeposit }: Pick<Cardano.ProtocolParameters, 'stakeKeyDeposit' | 'poolDeposit'>,
   certificates: Cardano.Certificate[],
-  rewardAccounts: Cardano.RewardAccount[]
+  rewardAccounts: Cardano.RewardAccount[] = []
 ): { deposit: Cardano.Lovelace; reclaimDeposit: Cardano.Lovelace } => {
-  if (rewardAccounts.length === 0 || certificates.length === 0) return { deposit: 0n, reclaimDeposit: 0n };
```

Comment:
@AngelCastilloB, what is the intent of `rewardAccounts.length === 0`?
Is it to avoid collecting/reclaiming deposits when the user has no reward accounts?

Reply:
This function (computeImplicitCoin) is used by the input selector and our util to compute transaction summary/display. The input selector doesn't have reward accounts to provide, but because we are building the transaction internally we know all the certificates are ours, so in this case the input selector pass an empy/undefined array to this function. However the transaction summary/display is receiving a transaction from a dApp, and can have mixed certificates (foreign and ours), so we need the list of reward accounts to be able to distinguish which deposits are going to our rewardaccounts from the ones that could potentially go to a different reward accounts that we dont control (same with reclaims)

Reply:
Nice summary, maybe worth putting it as a comment in code?

Reply:
done here [3a4da4f](https://github.com/input-output-hk/cardano-js-sdk/pull/1069/commits/3a4da4fa07dacc071d0f3cd0e567333f491c261a)